### PR TITLE
Add soft limit on allowed number of script fields in request

### PR DESCRIPTION
**Pull request by cbuescher**
Requesting to many script_fields in a search request can be costly
because of script execution. This change introduces a soft limit on the number
of script fields that are allowed per request. The setting can be
changed per index using the index.max_script_fields setting.

Relates to #26390

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/index/IndexSettings.java [hunk 1]
```diff
@@ -98,6 +98,15 @@
...
     public static final Setting<Integer> MAX_INNER_RESULT_WINDOW_SETTING =
         Setting.intSetting("index.max_inner_result_window", 100, 1, Property.Dynamic, Property.IndexScope);
+
+    /**
+     * Index setting describing the maximum value of allowed `script_fields`that can be retrieved
+     * per search request. The default maximum of 50 is defensive for the reason that retrieving
+     * script fields is a costly operation.
+     */
+    public static final Setting<Integer> MAX_SCRIPT_FIELDS_SETTING =
+        Setting.intSetting("index.max_script_fields", 50, 0, Property.Dynamic, Property.IndexScope);
```
Comment (cbuescher) [inline core/src/main/java/org/elasticsearch/index/IndexSettings.java]: note: 50 seems high enough for me, do you agree?

**Comment by cbuescher** [comment]
@jpountz similar to #26574, biggest question I have here is also about which versions to backport it to. I'd currently lean to merging this and similar soft limit changes to 6.0, 6.x and 7.0 to not having to introduce a change in request behaviour in a minor release. That would mean similar soft limit checks like #26492 should also be backported to 6.0? Whats your opinion on this?

File: core/src/main/java/org/elasticsearch/index/IndexSettings.java (hunk 1 above)
Comment (s1monw) [reply core/src/main/java/org/elasticsearch/index/IndexSettings.java]: I'd even go lower here maybe 32?

**Review by s1monw — APPROVED** [review]
LGTM

**Pull request merged**